src/common/async/context_pool.h

```diff
@@ -46,9 +47,14 @@ class io_context_pool {
   }
 public:
   io_context_pool() noexcept {}
+
   io_context_pool(std::int64_t threadcnt) noexcept {
     start(threadcnt);
   }
+  template<std::invocable<> Init>
+  io_context_pool(std::int64_t threadcnt, Init&& init = [] {}) noexcept {
```

Comment:
does the default work here? if so, we should be able to drop the init-less overloads